uftrace.c

```diff
@@ -555,6 +558,16 @@ static int parse_option(struct opts *opts, int key, char *arg)
 		opts->hide = opt_add_string(opts->hide, arg);
 		break;
 
+	case 'L':
+		pos = strchr(arg, '@');
+		if (!pos)
+			opts->loc_filter = opt_add_string(opts->loc_filter, arg);
```

Comment:
It'd be nice if we could add a warning for the invalid use - like for library path by old users.

I think we can check if the given path is a directory and has 'libmcount' subdirectory or 'libmcount.so' file.

Reply:
I added it. please check the conditions.

Reply:
I updated it, but it may not be a good way.
Please review it.
Thank you.

Reply:
Actually I got an idea.  So we want to check this only if short option `-L` is used, if they used `--library-path` then it should be fine.  Maybe we can separate it from the `--loc-filter` for a while so that existing uses can be warned.  How about adding a dummy (undocumented) option just for warning?

```c
static const struct option uftrace_options[] = {
	REQ_ARG(libmcount-path, OPT_libmcount_path),
	REQ_ARG(library-path, OPT_libmcount_path),
 	...
	REQ_ARG(loc-filter, OPT_loc_filter),
	REQ_ARG(loc-filter-warning, 'L'),  // the long option is dummy, will change later
	...
};

static int parse_option(struct opts *opts, int key, char *arg)
{
	switch (key) {
	case OPT_libmcount_path:
		/* ... */
	case 'L':
		if (is_libmcount_directory(arg))
			pr_warn(...);
		/* fall through */
	case OPT_loc_filter:
		/* ... */
```